2.6/root/usr/share/container-scripts/mongodb/common.sh

```diff
@@ -183,6 +183,65 @@ function setup_default_datadir() {
   fi
 }
 
+# check_env_vars checks environmental variables
+# if variables to create non-admin user are provided, sets CREATE_USER=1
+# if REPLICATION variable is set, checks also replication variables
+function check_env_vars() {
+  local readonly database_regex='^[^/\. "$]*$'
+
+  [[ -v MONGODB_ADMIN_PASSWORD ]] || usage "MONGODB_ADMIN_PASSWORD has to be set."
+
+  if [[ -v MONGODB_USER || -v MONGODB_PASSWORD || -v MONGODB_DATABASE ]]; then
+    [[ -v MONGODB_USER && -v MONGODB_PASSWORD && -v MONGODB_DATABASE ]] || usage "You have to set all or none of variables: MONGODB_USER, MONGODB_PASSWORD, MONGODB_DATABASE"
+
+    [[ "${MONGODB_DATABASE}" =~ $database_regex ]] || usage "Database name must match regexp: $database_regexp"
+    [ ${#MONGODB_DATABASE} -le 63 ] || usage "Database name too long (maximum 63 characters)"
+
+    CREATE_USER=1
+  fi
+
+  if [[ -v REPLICATION ]]; then
+    [[ -v MONGODB_KEYFILE_VALUE ]] || usage "MONGODB_KEYFILE_VALUE has to be set"
+    [[ -v MONGODB_REPLICA_NAME ]] ||  usage "MONGODB_REPLICA_NAME has to be set"
```

Comment:
i think this would be better as a single error so users don't enable replication, get the first error, set that variable, and then get the second error.